src/test/java/com/revature/controllertests/CurriculumModuleControllerTest.java

```diff
@@ -0,0 +1,170 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.*;
+import java.util.Set;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.revature.cmsforce.CMSforceApplication;
+import com.revature.controllers.CurriculumModuleController;
+import com.revature.entities.Curriculum;
+import com.revature.entities.CurriculumModule;
+import com.revature.entities.Link;
+import com.revature.entities.Module;
+import com.revature.entities.ReqLink;
+import com.revature.services.CurriculumModuleService;
+
+/** Test class for CRUD implementation of curriculum controller test. 
+ */
+
+@SpringBootTest(classes = CMSforceApplication.class)
+public class CurriculumModuleControllerTest {
+	
+	private MockMvc mvc;
+    private ObjectMapper objMapper = new ObjectMapper();
+    
+    @InjectMocks
+    private CurriculumModuleController cmc;
+    
+    @Mock
+    private CurriculumModuleService cms;
+    
+    @BeforeClass 
+    public void setup() {
+    	cmc = new CurriculumModuleController();
+    	mvc = MockMvcBuilders.standaloneSetup(cmc).build();
+    }
+    
+    @BeforeMethod
+    public void reinitMocks()
+    {
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    @Test
+    public void CreateCurriculumModule() throws Exception{
+    	Curriculum curriculum = new Curriculum(1,"Java Full Stack");
+    	Module module = new Module(1, "test_module", 1L, new HashSet<Link>(), new HashSet<ReqLink>(),
+                					new HashSet<Module>(), new HashSet<Module>());
+    	
+    	CurriculumModule curriculumModule = new CurriculumModule(1,curriculum.getId(),module,1);
+    	
+    	Mockito.when(cms.createCurrModule(curriculumModule)).thenReturn(curriculumModule);
+
+        ResultActions result = mvc.perform( post("/curriculummodules")
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(objMapper.writeValueAsString(curriculumModule)));
+
+        CurriculumModule actual = objMapper.readValue(result.andReturn().getResponse()
+                .getContentAsString(), CurriculumModule.class);
+
+        result.andExpect(status().isOk());
+        Mockito.verify(cms).createCurrModule(curriculumModule);
+        Assert.assertEquals(actual, curriculumModule, "Failed to retrieve expected curriculum module");
```

Comment:
Each test should have only one thing that could make it fail (expect ok  / assert method)

Reply:
+GetAllCurriculumModuleById +GetAllCurriculumModules

Reply:
moved to new branch  curriculum_feature_tests